barretenberg/cpp/src/barretenberg/ecc/fields/field_conversion.test.cpp

```diff
@@ -46,28 +47,46 @@ TEST_F(FieldConversionTest, FieldConversionGrumpkinFr)
     check_conversion(x1);
 }
 
+namespace {
+bb::fq derive_bn254_y(bb::fq x)
+{
+    auto [found, y] = (x.sqr() * x + Bn254G1Params::b).sqrt();
+    EXPECT_TRUE(found);
+    return y;
+}
+} // namespace
+
 /**
  * @brief Field conversion test for curve::BN254::AffineElement
  *
  */
 TEST_F(FieldConversionTest, FieldConversionBN254AffineElement)
 {
-    curve::BN254::AffineElement x1(1, 2);
+    curve::BN254::AffineElement x1(1, derive_bn254_y(1));
```

Comment:
oh wait, it's able to find the point for both x values? would not have expected it

Reply:
hmm indeed it should crap out if not